dask/bytes/http.py

```diff
@@ -0,0 +1,284 @@
+from __future__ import print_function, division, absolute_import
+
+import requests
+
+from . import core
+from ..base import tokenize
+
+DEFAULT_BLOCK_SIZE = 5 * 2 ** 20
+
+
+class HTTPFileSystem(core.FileSystem):
+    """
+    Simple File-System for fetching data via HTTP(S)
+
+    Unlike other file-systems, HTTP is limited in that it does not provide glob
+    or write capability. Furthermore, no read-ahead is presently
+    """
+    sep = '/'
+
+    def __init__(self, **storage_options):
+        """
+        Parameters
+        ----------
+        storage_options: key-value
+            May be credentials, e.g., `{'auth': ('username', 'pword')}` or any
+            other parameters for requests
+        """
+        self.block_size = storage_options.pop('block_size', DEFAULT_BLOCK_SIZE)
+        self.kwargs = storage_options
+        self.session = requests.Session()
+
+    def glob(self, url):
+        """For a template path, return matching files"""
+        raise NotImplementedError
+
+    def mkdirs(self, url):
+        """Make any intermediate directories to make path writable"""
+        raise NotImplementedError
+
+    def open(self, url, mode='rb', block_size=None, **kwargs):
+        """Make a file-like object
+
+        Parameters
+        ----------
+        url: str
+            Full URL with protocol
+        mode: string
+            must be "rb"
+        kwargs: key-value
+            Any other parameters, passed to requests calls
+        """
+        if mode != 'rb':
+            raise NotImplementedError
+        block_size = block_size if block_size is not None else self.block_size
+        return HTTPFile(url, self.session, block_size, **self.kwargs)
+
+    def ukey(self, url):
+        """Unique identifier, so we can tell if a file changed"""
+        # Could do HEAD here?
+        return tokenize(url)
```

Comment:
This seems dangerous if the underlying file changes.

Reply:
I think it can be an assumption of the FS that we're dealing with static files. The server is not guaranteed to provide ETAGs or any other checksums, timestamp or even size, so I don't think there's anything else we can do (except make this a uuid and reload every time).

Reply:
Some people use Dask over an extended time.  I think that they would find this behavior surprising.  I'm inclined to use uuids.